src/globus_sdk/services/transfer/client.py

```diff
@@ -5,17 +5,19 @@
 
 from globus_sdk import client, exc, paging, response, utils
 from globus_sdk.scopes import TransferScopes
+from globus_sdk.types import ToStr
 
 from .data import DeleteData, TransferData
 from .errors import TransferAPIError
 from .response import ActivationRequirementsResponse, IterableTransferResponse
 
 log = logging.getLogger(__name__)
 
+# REVIEW: can this be ToStr?
 ID_PARAM_TYPE = Union[bytes, str, uuid.UUID]
```

Comment:
My response to this is that I want to be more certain that `ToStr` gives a similar level of safety. (See my comment there, about the default `__str__`.)

For example, would `ToStr` prevent this?
```python
epid = "foo",  # whoops, you made a tuple!
tc.get_endpoint(epid)
```

Reply:
Pointing to this as a specific example, let me return to this question with a better phrasing: if we make `ToStr` a more specific interface, perhaps a slight superset of `ID_PARAM_TYPE` here, does it capture the original purpose of `ID_PARAM_TYPE` to use `ToStr` instead?

Reply:
I think the thing to compare against is something like `server_id`, which is an integer.
Without worrying about generalizing, these are probably the desirable types:
```
server_id: int | str | bytes
endpoint_id: UUID | str | bytes
```

So there's no one type we can write which will express this for us. Why not just write them down as separate types?
We could define these globally, e.g. `from globus_sdk.types import INT_T, UUID_T`.

Reply:
So, "this type or a string of it"? And then probably `ToStr` is no longer needed. What about `IntLike`/`UUIDLike`?

Reply:
Yeah, I think that's how we want to think about this: "an integer, or a string of it; a uuid, or a string of it; etc".

I'm cool with those names for the types. `IntLike` seems clear to me.

I'll go ahead and ask, since it seems related: do we think supporting `bytes` is even important? If we discard that, we could probably replace all calls to `safe_stringify` with just `str()`. (There's a note about string duplication in that function, but I think that was a python2 behavior. `str()` of a string just returns the string now.)

I know I mentioned that there are python interfaces which handle bytes, but dropping support for this from the SDK would just mean that users with unencoded bytes have to call `encode()` themselves. It's not a big deal for them and it's a simpler surface for us to maintain.